### ISSUE TITLE
MIPS "Invalid register name global variable" with `__asm__("sp")`

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#1608]
After commit [200ed341b864](https://git.kernel.org/mips/c/200ed341b8646ea893d5466974d6e107c46237ba) ("mips: Implement "current_stack_pointer"") in -next:

```
$ make -skj"$(nproc)" ARCH=mips LLVM=1 mrproper malta_defconfig all
...
fatal error: error in backend: Invalid register name global variable
...
```

Looking at [`MipsTargetLowering::getRegisterByName()`](https://github.com/llvm/llvm-project/blob/44eee659f1c530a684fa2e57a983d962b5980620/llvm/lib/Target/Mips/MipsISelLowering.cpp#L4720-L4741) in `llvm/lib/Target/Mips/MipsISelLowering.cpp`, this appears expected.

I have opened an upstream LLVM issue: https://github.com/llvm/llvm-project/issues/54174

**Comment by nickdesaulniers** [comment]
https://reviews.llvm.org/D120926

**Comment by nickdesaulniers** [comment]
Now to figure what to do on the kernel side.

**Comment by kees** [comment]
Kernel patch for earlier Clang versions sent:

https://lore.kernel.org/lkml/20220309204537.390428-1-keescook@chromium.org

**Comment by nathanchance** [comment]
This should be all fixed up as of `next-20220315`: https://git.kernel.org/next/linux-next/c/b847bd64ea9f484510e27065cb2bccc58d9b829b
